### DISCUSSION TITLE
Can i use SSE to send message to user without saving or updating a collection?

### DISCUSSION THREAD

**Discussion by jordant97** [pocketbase/pocketbase#5319] [Q&A] [answered]
I wanted to implement a chat feature in the project but don't need the messages to be saved. Is it possible to use the SSE in PocketBase to just broadcast the messages? My thinking is that every user will listen to changes on a roomID to establish the real-time connection and use that as the reference for IDs to broadcast the messages. Any help is appreciated, thanks for this great software

**Comment by ganigeorgiev** [comment, marked as answer]
You can but only programmatically when using Go (or the JSVM). Since your message is not associated to a collection you'll have to [register a custom route](https://pocketbase.io/docs/go-routing/#registering-new-routes) that will accept the specific payload and then you can send it to all [`app.SubscriptionsBroker().Clients()`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.18/tools/subscriptions#Broker):

- Go :
    ```go
    message := subscriptions.Message{
        Name: "example",             // the subscription topic
        Data: []byte(`{"abc":123}`), // any string
    }

    clients := app.SubscriptionsBroker().Clients()
    for _, c := range clients {
        if c.HasSubscription(message.Name) {
            // you can wrap it in routine.FireAndForget to execute in a separate goroutine
            c.Send(message)
        }
    }
    ```
    _(with Go you have more flexibility and control over the execution)_
    
- JS `pb_hooks`:
   
    ```js
    const message = new SubscriptionMessage({
        name: "example",
        data: '{"abc": 123}',
    })

    const clients = $app.subscriptionsBroker().clients()
    for (let id in clients) {
        const client = clients[id]
        if (client.hasSubscription(message.name)) {
            client.send(message)
        }
    }
    ```

To subscribe to the messages on the client-side, with the JS SDK you can use the [`pb.realtime` service](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#realtimeservice):

```js
pb.realtime.subscribe("example", (e) => {
  ...
})
```

To send a request to your custom route with the payload you can make use of the `pb.send()` method() (_see [Sending request to custom routes using the SDKs](https://pocketbase.io/docs/go-routing/#sending-request-to-custom-routes-using-the-sdks)_).
